### PR TITLE
Add --config option

### PR DESCRIPTION
**Pull request by dg10a**
Addresses https://github.com/emersion/mako/issues/171

Not sure if there are issues with looping over the arguments twice. Feedback welcome.

### CODE REVIEW

**Comment by emersion** [comment]
Ah, interesting, someone sent me a very similar patch some time ago: https://lists.sr.ht/~emersion/public-inbox/patches/8076

**Comment by dg10a** [comment]
Glad to see it's being worked on. Let me know if you want me to close this PR in favor of that patch.

**Comment by emersion** [comment]
I'd be interested to see an approach that addresses the concerns explained there: https://lists.sr.ht/~emersion/public-inbox/patches/8076#%3CtpSpmtft2kY55ktdn17GJ7UCRZHWb7Qxpr97WfHwMEcVL51_ighZLa50bdYLZcsMCn1sRHXKFJvpfpsqj4qy-9svQpg6V3snyDNB1ZhvAKc=@emersion.fr%3E

**Comment by dg10a** [comment]
I think the current PR meets your concerns, but it loops with `getopt` twice. What do you think of the approach below? It works neatly for the style options, but I'm not sure how to handle the config options.

```C
// store style options
struct mako_style opt_style;
init_empty_style(&opt_style);

optind = 1;
char *config_arg = NULL;
while (1) {
	int option_index = -1;
	int c = getopt_long(argc, argv, "hc:", long_options, &option_index);
	if (c < 0) {
		break;
	} else if (c == 'h') {
		return 1;
	} else if (c == 'c') {
		config_arg = strdup(optarg);
	} else if (c != 0) {
		return -1;
	} else {
		const char *name = long_options[option_index].name;
		if (!apply_style_option(&opt_style, name, optarg)
				&& !apply_config_option(config, name, optarg)) {
			fprintf(stderr, "Failed to parse option '%s'\n", name);
			return -1;
		}
	}
}

int config_status = load_config_file(config, config_arg);
if (config_status < 0) {
	return -1;
}

// apply style options
apply_style(&root_criteria->style, &opt_style);
finish_style(&opt_style);
```

File: config.c [hunk 1]
```diff
@@ -544,10 +544,15 @@ static char *get_config_path(void) {
 	return NULL;
 }
 
-int load_config_file(struct mako_config *config) {
-	char *path = get_config_path();
-	if (!path) {
-		return 0;
+int load_config_file(struct mako_config *config, char* config_arg) {
```
Comment (emersion) [inline config.c]: Style: `*` on the right

**Pull request merged**